psa_builder.py

```diff
@@ -237,7 +236,32 @@ def check_and_clone_repo(name, deps, dir):
         logging.info(
             "%s repo exists, fetching latest from remote %s", name, deps
         )
-        cmd = ["git", "-C", join(dir, name), "fetch", deps]
+        cmd = [
+            "git",
+            "-C",
+            tfm_dir,
+            "remote",
+            "get-url",
+            deps,
+        ]
+        ret = run_cmd_and_return(cmd)
+        if ret != 0:
+            logging.info("%s is not a remote, adding it", deps)
+            cmd = [
+                "git",
+                "-C",
+                tfm_dir,
+                "remote",
+                "add",
+                deps,
+                dependencies[deps].get(name)[0],
+            ]
+            ret = run_cmd_and_return(cmd)
+            if ret != 0:
```

Comment:
We could return a boolean or define a `Enum` with possible return codes here. In Python you want to probably use a higher level construct than an int to signify a failure. We could even use exceptions... 

How do we know just from the return code what the actual error was? We assume it's a failure to add a remote or a failure to find a remote. Could we make this more explicit by either raising a specific exception or returning a more specific error code?

Reply:
Yes, we should use exceptions more than we do. Here, we don't care what the error is very much. If anything failed, we explain that we failed what we attempted to do. This can be unhelpful as we aren't explaining _why_ we failed.

Reply:
The error code and exception stuff is probably covered by #104. I'm assuming we can't explain why we failed until the error handling is reworked so this function produces more specific return codes or exceptions.